R/extract_fin_year.R

```diff
@@ -18,8 +18,8 @@
 #' extract_fin_year(x)
 #' @export
 extract_fin_year <- function(date) {
-  if (!inherits(date, c("Date", "POSIXct"))) {
-    cli::cli_abort("{.arg date} must be a {.cls Date} or {.cls POSIXct} vector,
+  if (!inherits(date, c("Date", "POSIXt"))) {
```

Comment:
Any reason we can't accept POSIXt and POSIXct? I'm never really sure what the difference is...

Potentially changing this line will break existing code but ideally this function would just accept any 'reasonable' date type and deal with it without complaint!

Reply:
I believe `POSIXt` is a "lower" class than `POSIXct` meaning that by checking if it's a `POSIXt`, it will accept both `POSIXct` and `POSIXlt` classes. The former version did not accept `POSIXlt` which makes no sense as `lubridate::year()` calls `as.POSIXlt()` haha. 
To clarify: my suggested version accepts `POSIXlt`, `POSIXct`, and anything with a `POSIXt` class.

Reply:
``` r
class(Sys.time())
#> [1] "POSIXct" "POSIXt"
class(as.POSIXlt(Sys.time()))
#> [1] "POSIXlt" "POSIXt"
```

<sup>Created on 2023-11-20 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>